src/app/tests/suites/include/ConstraintsChecker.h

```diff
@@ -269,6 +269,24 @@ class ConstraintsChecker
         return CheckConstraintMaxValue(itemName, current.Value(), static_cast<T>(expected));
     }
 
+    template <typename T>
+    bool CheckConstraintNotValue(const char * itemName, const chip::app::DataModel::Nullable<T> & current,
+                                 const chip::app::DataModel::Nullable<T> & expected)
+    {
+        if (expected.IsNull() && current.IsNull())
+        {
+            Exit(std::string(itemName) + " got NULL for both values, but expected not equal");
+            return false;
+        }
+
+        if ((expected.IsNull() && !current.IsNull()) || (!expected.IsNull() && current.IsNull()))
```

Comment:
```suggestion
        if (expected.IsNull() != current.IsNull())
```

Reply:
Yes, this is much more simple and clear. Updated.